ush/python/pygfs/task/atmens_analysis.py

```diff
@@ -0,0 +1,332 @@
+from pygw.fsutils import rm_p, chdir
+from pygw.yaml_file import parse_yamltmpl, parse_j2yaml, save_as_yaml
+from pygw.logger import logit
+from pygw.executable import Executable
+from pygw.exceptions import WorkflowException
+from pygfs.task.analysis import Analysis
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class AtmEnsAnalysis(Analysis):
+    """
+    Class for global atmens analysis tasks
+    """
+    @logit(logger, name="AtmEnsAnalysis")
+    def __init__(self, config):
+        super().__init__(config)
+
+        _res = int(self.config.CASE_ENKF[1:])
+        _res_anl = int(self.config.CASE_ANL[1:])
+        _window_begin = add_to_datetime(self.runtime_config.current_cycle, -to_timedelta(f"{self.config.assim_freq}H") / 2)
+        _fv3jedi_yaml = os.path.join(self.runtime_config.DATA, f"{self.runtime_config.CDUMP}.t{self.runtime_config.cyc:02d}z.atmens.yaml")
+
+        # Create a local dictionary that is repeatedly used across this class
+        local_dict = AttrDict(
+            {
+                'npx_ges': _res + 1,
+                'npy_ges': _res + 1,
+                'npz_ges': self.config.LEVS - 1,
+                'npz': self.config.LEVS - 1,
+                'npx_anl': _res_anl + 1,
+                'npy_anl': _res_anl + 1,
+                'npz_anl': self.config.LEVS - 1,
+                'ATM_WINDOW_BEGIN': _window_begin,
+                'ATM_WINDOW_LENGTH': f"PT{self.config.assim_freq}H",
+                'comin_ges_atm': self.config.COMIN_GES,
+                'comin_ges_atmens': self.config.COMIN_GES_ENS,
+                'OPREFIX': f"{self.config.EUPD_CYC}.t{self.runtime_config.cyc:02d}z.",  # TODO: CDUMP is being replaced by RUN
+                'APREFIX': f"{self.runtime_config.CDUMP}.t{self.runtime_config.cyc:02d}z.",  # TODO: CDUMP is being replaced by RUN
+                'GPREFIX': f"gdas.t{self.runtime_config.previous_cycle.hour:02d}z.",
+                'fv3jedi_yaml': _fv3jedi_yaml,
+            }
+        )
+
+        # task_config is everything that this task should need
+        self.task_config = AttrDict(**self.config, **self.runtime_config, **local_dict)
+
+    @logit(logger)
+    def initialize(self: Analysis) -> None:
+        """Initialize a global atmens analysis
+
+        This method will initialize a global atmens analysis using JEDI.
+        This includes:
+        - staging CRTM fix files
+        - staging FV3-JEDI fix files
+        - staging model backgrounds
+        - generating a YAML file for the JEDI executable
+        - linking the JEDI executable (TODO make it copyable, requires JEDI fix)
+        - creating output directories
+
+        Parameters
+        ----------
+        Analysis: parent class for GDAS task
+
+        Returns
+        ----------
+        None
+        """
+        super().initialize()
+
+        # stage CRTM fix files
+        crtm_fix_list_path = os.path.join(self.task_config.HOMEgfs, 'parm', 'parm_gdas', 'atm_crtm_coeff.yaml')
```

Comment:
I notice this repeats from `atm_analysis.py`.

Reply:
Not sure if this is what you have in mind @aerorahul , but here is what I did in a working copy of `ush/python/pygfs/task/analysis.py` and `ush/python/pygfs/task/atmens_analysis.py`

The following was added to the `initialize` method in `analysis.py`
```
         obs_dict = self.get_obs_dict()
         FileHandler(obs_dict).sync()

+        # all analyses need to stage fix files
+        jedi_fix_list_path = os.path.join(self.task_config.HOMEgfs, 'parm', 'parm_gdas', 'atm_jedi_fix.yaml')
+        logger.debug(f"Staging JEDI fix files from {jedi_fix_list_path}")
+        jedi_fix_list = parse_yamltmpl(jedi_fix_list_path, self.task_config)
+        FileHandler(jedi_fix_list).sync()
+
+        # some analyses need to stage crtm files
+        crtm_fix_list_path = os.path.join(self.task_config.HOMEgfs, 'parm', 'parm_gdas', 'atm_crtm_coeff.yaml')
+        logger.debug(f"Staging CRTM fix files from {crtm_fix_list_path}")
+        crtm_fix_list = parse_yamltmpl(crtm_fix_list_path, self.task_config)
+        FileHandler(crtm_fix_list).sync()
+
         # some analyses need to stage bias corrections
         bias_dict = self.get_bias_dict()
         FileHandler(bias_dict).sync()
```
The following was deleted from `atmens_analysis.py`
```
         super().initialize()

-        # stage CRTM fix files
-        crtm_fix_list_path = os.path.join(self.task_config.HOMEgfs, 'parm', 'parm_gdas', 'atm_crtm_coeff.yaml')
-        logger.debug(f"Staging CRTM fix files from {crtm_fix_list_path}")
-        crtm_fix_list = parse_yamltmpl(crtm_fix_list_path, self.task_config)
-        FileHandler(crtm_fix_list).sync()
-
-        # stage fix files
-        jedi_fix_list_path = os.path.join(self.task_config.HOMEgfs, 'parm', 'parm_gdas', 'atm_jedi_fix.yaml')
-        logger.debug(f"Staging JEDI fix files from {jedi_fix_list_path}")
-        jedi_fix_list = parse_yamltmpl(jedi_fix_list_path, self.task_config)
-        FileHandler(jedi_fix_list).sync()
-
         # stage backgrounds
         FileHandler(self.get_bkg_dict(AttrDict(self.task_config))).sync()
```
Tests in the prufsatm show that the above changes work.

Reply:
I looked in `parm/parm_gdas`.  I see that we have different crtm_coeff yamls for UFS-DA Aerosol and ATM.  My attempt to move the staging of crtm files from `atmens_analysis.py` to `analysis.py` is not sufficiently general.   It works for UFS-DA ATM and ENS but not Aerosol.

Reply:
The same comment applies for jedi_fix.    `aero_jedi_fix.yaml` and `atm_jedi_fix.yaml` reference different fieldmetadata files.    The jedif_fix change to my working copy of `analysis.py` only works for UFS-DA ATM and ENS.  It won't work for UFS-DA Aerosol.

Reply:
Recommend that unification of fix file staging across UFS-DA applications be the subject of a new g-w issue.   What do you think @aerorahul and @CoryMartin-NOAA ?

Reply:
I think I know what to do.  We can continue with the current approach in this PR.
We can revisit this in a future PR.  I may request assistance from @RussTreadon-NOAA @CoryMartin-NOAA to test since I haven't quite gotten a working setup for UFS-DA.  It will happen.